dist/react-maskedinput.js

```diff
@@ -42,6 +42,25 @@ var MaskedInput = React.createClass({displayName: "MaskedInput",
     })
   },
 
+  componentWillUpdate:function(nextProps, nextState) {
+    if (nextProps.pattern !== this.props.pattern) {
+      this._updatePattern(nextProps)
+    }
+  },
+
+  componentDidUpdate:function(prevProps) {
+    if (prevProps.pattern !== this.props.pattern && this.mask.selection.start) {
+      this._updateInputSelection()
+    }
+  },
+
+  _updatePattern: function(props) {
+    this.mask.setPattern(props.pattern, {
+      value: this.props.value,
```

Comment:
I think this should use `this.mask.getRawValue()` - this way it's not going to put placeholders where chars should be when the pattern changes.